### PR TITLE
React-ui: Exposed status prop in useAuth

### PR DESCRIPTION
**Pull request by jmderby**
## Description

Expose `status` prop in `useAuth` hook to enable better loading UX in applications. This change allows developers to create more responsive interfaces by providing granular control over authentication states. The PR includes updates to the demo showcasing improved loading states and adds a test case for the new prop.

## Test plan

Ensure demo still works as usual and all loading UX is suffice.

## Package updates

**@crossmint/client-sdk-react-ui** 
- added prop 'status' to useAuth hook
- updated test case

### CODE REVIEW

File: apps/wallets/smart-wallet/next/src/app/page.tsx [hunk 1]
```diff
@@ -49,20 +81,7 @@ export default function Home() {
...
-                            >
-                                Open in my wallet
-                            </Link>
-                        </div>
-                    ) : null}
-                    {walletStatus === "loaded" && !nftSuccessfullyMinted ? (
-                        <MintNFTButton setNftSuccessfullyMinted={setNftSuccessfullyMinted} />
-                    ) : null}
-                    {walletStatus !== "loaded" && !nftSuccessfullyMinted ? <SignInAuthButton /> : null}
+                    {renderWalletContent()}
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/page.tsx:84]: i should rename this function, any ideas?

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx [hunk 2]
```diff
@@ -158,17 +155,41 @@ describe("CrossmintAuthProvider", () => {
...
     });
+
+    test(`Logging in and asserting the auth status`, async () => {
+        const { getByTestId } = renderAuthProvider({
+            children: <TestComponent />,
+            embeddedWallets,
+        });
+
+        // Because our test defaults the jwt to "mock-jwt", we need to clear it first
+        fireEvent.click(getByTestId("clear-jwt-button"));
```
Comment (devlyn37) [inline packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.test.tsx:181]: You could also change the mock here instead

File: packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx
Comment (devlyn37) [inline packages/client/ui/react-ui/src/providers/CrossmintAuthProvider.tsx:1]: If possible we should derive `status` from existing state!

File: apps/wallets/smart-wallet/next/src/app/page.tsx [hunk 3]
```diff
@@ -14,6 +15,37 @@ export default function Home() {
     const { status: walletStatus } = useWallet();
     const [nftSuccessfullyMinted, setNftSuccessfullyMinted] = useState(false);
 
+    const renderActionButton = () => {
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/app/page.tsx:18]: Screenshot 2024-09-08 at 9 02 51 PM

I'm not sure this is working as intended

**Review by devlyn37 — COMMENTED** [review]
Screenshot 2024-09-08 at 9 02 51 PM

I think something is busted in the demo.

**Comment by jmderby** [comment]
Closing PR because moving demo changes to another PR

**Pull request closed without merging**